src/lib/Libifl/pbsD_connect.c

```diff
@@ -592,7 +623,45 @@ __pbs_disconnect(int connect)
 	if (pbs_client_thread_destroy_connect_context(connect) != 0)
 		return -1;
 
-	(void)destroy_connection(connect);
+	destroy_connection(connect);
+
+	return 0;
+}
+
+/**
+ * @brief
+ *	-send close connection batch request
+ *
+ * @param[in] connect - socket descriptor
+ *
+ * @return	int
+ * @retval	0	success
+ * @retval	-1	error
+ *
+ */
+int
+__pbs_disconnect(int connect)
```

Comment:
so pbs_connect() will automatically connect to N underlying server instances, but pbs_disconnect() will disconnect only from the one exact fd (connect) passed to it. That sounds like a disparity in the design, no?

In other words, which call, or what party would know to call pbs_disconnect() on each of the underlying sever instance connections?

Reply:
good point, we should disconnect from all if we are connecting to all inside pbs_connect